src/encoded/static/components/facets/defaults.js

```diff
@@ -763,6 +763,88 @@ DefaultDateSelectorFacet.contextTypes = {
+        initializeBiochemicalInput = initializeBiochemicalInput.concat(['All']);
+    }
+    const [biochemicalInput, setBiochemicalInput] = React.useState(initializeBiochemicalInput);
+
+    // Select or de-select biochemical input terms
+    const handleClick = React.useCallback((event) => {
+        const newInput = event.target.getAttribute('id').toString();
+        const newInputIdx = biochemicalInput.indexOf(newInput);
+        const query = new QueryString(queryString);
+        // New term is "All" and "All" is already selected so we want to de-select all biochemical inputs
+        if (newInput === 'All' && (newInputIdx > -1)) {
+            setBiochemicalInput([]);
+            query.deleteKeyValue('biochemical_inputs');
+        // New term is "All" and "All" is not already selected so we want to select all biochemical inputs
+        } else if (newInput === 'All' && (newInputIdx === -1)) {
+            facet.terms.forEach((term) => {
+                if (biochemicalInput.indexOf(term.key) === -1) {
+                    query.addKeyValue('biochemical_inputs', term.key);
+                }
+            });
+            const allBiochemicalInputs = facet.terms.map((term) => term.key).concat(['All']);
+            setBiochemicalInput(allBiochemicalInputs);
+        // "All" is not selected and term already is in list so remove term
+        } else if (newInputIdx > -1) {
+            const biochemicalInputCopy = biochemicalInput.filter((input) => input !== newInput);
+            setBiochemicalInput(biochemicalInputCopy);
+            query.deleteKeyValue('biochemical_inputs', newInput);
+        // "All" is not selected and term is not in list so add term
+        // Check to see if this means all terms are selected
+        } else {
+            let biochemicalInputCopy = biochemicalInput.concat([newInput]);
+            if (biochemicalInputCopy.length === facet.terms.length) {
+                biochemicalInputCopy = biochemicalInputCopy.concat(['All']);
+            }
+            setBiochemicalInput(biochemicalInputCopy);
+            query.addKeyValue('biochemical_inputs', newInput);
+        }
+        const href = `?${query.format()}`;
+        reactContext.navigate(href);
+    });
+
+    return (
+        <div className="facet">
+            <h5>{facet.title}</h5>
+            <div className="facet__multiselect">
+                {facet.terms.sort((a, b) => (a.key).localeCompare(b.key)).map((term) => (
+                    <button className={(biochemicalInput.indexOf(term.key) > -1) ? 'selected' : ''} key={term.key} id={term.key} onClick={handleClick} type="button">
+                        {term.key}
+                    </button>
+                ))}
+                <button className={(biochemicalInput.indexOf('All') > -1) ? 'selected' : ''} key="All" id="All" onClick={handleClick} type="button">
+                    All
+                </button>
+            </div>
+        </div>
+    );
+};
```

Comment:
I've been thinking about using hrefs instead of the click handler and I'm not sure if it will be simpler, but maybe I'm missing something. I think I would have to check the url to see if the term is already present, and add it if not, and then check the url for all of the terms for the "All" link/button and construct the url based on that. This seems like I would be to some extent duplicating code / logic that already exists in the QueryString library? Maybe I'm over-thinking this or missing something though. I was thinking you were suggesting using location_href and doing string parsing.

Reply:
> I think I would have to check the url to see if the term is already present, and add it if not, and then check the url for all of the terms for the "All" link/button and construct the url based on that

That’s what I was thinking.

> This seems like I would be to some extent duplicating code / logic that already exists in the QueryString library?

I think you could _use_ the QueryString library to parse and modify the query string for each button. The `getKeyValues` method can tell you all the values in every “biochemical_input=” query-string parameter, and using that to determine which buttons get highlighted, along with (probably cloning and) updating the query-string object to generate the new links for each button.

Reply:
Ok, well I re-wrote it, hopefully this is sort of along the lines of what you were thinking!

Since all the buttons are all now links but we wanted to keep the button styles, I had to add some styles and I think I resolved the Safari border issue which I had forgotten to address previously.